packages/tldraw/src/lib/ui/components/Toolbar/OverflowingToolbar.tsx

```diff
@@ -0,0 +1,199 @@
+import { preventDefault, useEditor, useEvent } from '@tldraw/editor'
+import classNames from 'classnames'
+import hotkeys from 'hotkeys-js'
+import { createContext, useEffect, useId, useLayoutEffect, useMemo, useRef, useState } from 'react'
+import { PORTRAIT_BREAKPOINT } from '../../constants'
+import { useBreakpoint } from '../../context/breakpoints'
+import { areShortcutsDisabled } from '../../hooks/useKeyboardShortcuts'
+import { TLUiToolItem } from '../../hooks/useTools'
+import { useTranslation } from '../../hooks/useTranslation/useTranslation'
+import { TldrawUiButton } from '../primitives/Button/TldrawUiButton'
+import { TldrawUiButtonIcon } from '../primitives/Button/TldrawUiButtonIcon'
+import {
+	TldrawUiDropdownMenuContent,
+	TldrawUiDropdownMenuRoot,
+	TldrawUiDropdownMenuTrigger,
+} from '../primitives/TldrawUiDropdownMenu'
+import { TldrawUiMenuContextProvider } from '../primitives/menus/TldrawUiMenuContext'
+
+export const IsInOverflowContext = createContext(false)
+
+export function OverflowingToolbar({ children }: { children: React.ReactNode }) {
+	const editor = useEditor()
+	const id = useId().replace(/:/g, '_')
```

Comment:
why the replacement here? i see we do it in ArrowShapeUtil - should there be a util function `safeId` or something like that? hard to know why this is necessary on first read